py3status/modules/gazelle.py

```diff
@@ -0,0 +1,164 @@
+# -*- coding: utf-8 -*-
+"""
+Show your ratio on your favorite Gazelle-based private tracker.
+
+This module can get various useful infos about you on a private tracker.
+You can display your ratio, upload size, download size.
+The only requirements are that the private tracker is based on the Gazelle framework,
+for instance Apollo or Redacted, and that the sysops didn't block API access somehow.
+This module will save session cookies to avoid too much opened sessions which could
+potentially put you into troubles due to a suspect activity.
+
+Configuration parameters:
+    baseurl: Base URL of the tracker (example: https://what.cd) (default None)
+    cache_timeout: How often we refresh this module in seconds (default 60)
+    format: See placeholders below (default 'Ratio: {ratio}')
+    password: Account password (default '')
+    thresholds: specify color thresholds to use
+        (default [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')])
+    urgent_new_messages: set output urgent mode when there are new messages
+        (default False)
+    urgent_new_notifs: set output urgent mode when there are new notifications
+        (default False)
+    username: Account username (default '')
+
+Format placeholders:
+    {ratio} account ratio
+    {download} account download amount
+    {upload} account upload amount
+    {messages} number of unread messages
+    {notifs} number of new notifications
+
+@author raspbeguy <raspbeguy@hashtagueule.fr>
+@license MIT
+"""
+
+from py3status.exceptions import RequestException
+
+import re
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    baseurl = None
+    cache_timeout = 60
+    format = 'Ratio: {ratio}'
+    password = ''
+    thresholds = [(0, 'bad'), ('requiredratio', 'degraded'), (1, 'good')]
+    urgent_new_messages = False
+    urgent_new_notifs = False
```

Comment:
Thoughts on using per-module configuration `allow_urgent = False` to turn them off? I assume users won't get too much messages / notifications to have a need to turn one off? I'm wondering if this is a superflous config and if we can avoid it... Also, Idk the diff between `messages` and `notifications`.

Reply:
* Notifications: On the tracker frontend, you can define some torrents filters. Notifications are the numbers of unseen torrents that match those filters. Frequency is variable depending of your filters.
* Messages: private messages between users. Frequency is kind of low, consider it like a spam-free personal mailbox.

Because those two counters are very different, we need to differentiate the urgent behavior for each of them.

Reply:
Maybe `use_urgent = ['messages', 'notifs']` instead.

Reply:
what's the purpose of that, except making the config even uglier?

Reply:
Idk. None other than simplifying and/or aiming for minimal number of configs. This is your baby and I'm just messing around with it. Based on your description, the sensible default could be just `use_urgent = ['messages']` as `notifs` could be too noisy (and/or too repeatedly). IIRC we can't use `urgent = []` because it's a reserved word.

Ugly config? Huh... I thought it looks nice. Maybe in the future, we could have `[\?if=messages&urgent ]`... Ugly but could be very flexible with urgent. Idk if that'll even happen. :-) Anyhow, I defer this suggestion to you/others.

Maybe it's better to just work with `messages` (it's more of a good purpose for `urgent` rather than noisy `notifis`)... then we might not need those `urgent` configs.

Reply:
I really don't see the need for making it a list instead of 2 separate vars. It seems way clearer for me. Anyway I would like to avoid using python syntax in a config file as much as possible (that also explain my hate for the thresholds system).

If notifs are too noisy, then the user will decide to turn `urgent` off for them. In my case, I want to get special event when I have some notifications, which aren't that noisy.